jsonnet/patch-rules.libsonnet

```diff
@@ -124,19 +328,70 @@ local patchedRules = [
 local patchOrExcludeRule(rule, ruleSet, operation) =
   if std.length(ruleSet) == 0 then
     [rule]
-  else if (('alert' in rule && 'alert' in ruleSet[0]) && std.startsWith(rule.alert, ruleSet[0].alert)) ||
-          (('record' in rule && 'record' in ruleSet[0]) && std.startsWith(rule.record, ruleSet[0].record)) then
-    if operation == 'patch' then
-      local patch = {
-        [k]: ruleSet[0][k]
-        for k in std.objectFields(ruleSet[0])
-        if k != 'alert' && k != 'record'
-      };
-      [std.mergePatch(rule, patch)]
+  else if ('alert' in rule) then
```

Comment:
If we stick to only updating the severity, we shouldn't have to update this `patchOrExcludeRule` function. The only improvement that we might need is to fix how it handles alerts that have the same name. My suggestion would be to introduce a hidden `index` field to the rules in `patchedRules` that would be skipped if not set. We could then select the rules we want to update without having to select the fields that differ between the rules. For instance, it could be the `severity` or the `expr` so instead of having to support all that, it is easier to add an index-based selection.